### PR TITLE
ASB-735: support multiple missions on stpubdata

### PR DESCRIPTION
**Pull request by cam72cam**
At STScI, we are gearing up to put TESS data on S3.  We would like to get the code in before the TESS data release go-live.

Access to the TESS data will be locked down until the official go live.  Here is a example program showcasing the output.  The product had to be constructed by hand as we do not have the products in our ops database yet.

In addition, we have added the ability to specify a [profile name](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html) for the cloud credentials configuration

```python
from astroquery.mast import Observations
products = [{
    "obs_collection": "TESS",
    "obs_id": "206409997",
    "dataURI": "mast:TESS/product/tess2018206045859-s0001-0000000206409997-0120-s_tp.fits",
    "productFilename": "tess2018206045859-s0001-0000000206409997-0120-s_tp.fits"
}]
Observations.enable_cloud_dataset()

manifest = Observations._download_files(products, "mastDownload")
```
```
INFO: Using the S3 STScI public dataset [astroquery.mast.core]
WARNING: Your AWS account will be charged for access to the S3 bucket [astroquery.mast.core]
INFO: See Request Pricing in https://aws.amazon.com/s3/pricing/ for details [astroquery.mast.core]
INFO: If you have not configured boto3, follow the instructions here: https://boto3.readthedocs.io/en/latest/guide/configuration.html [astroquery.mast.core]
Downloading URL s3://stpubdata/tess/public/tid/s0001/0000/0002/0640/9997/tess2018206045859-s0001-0000000206409997-0120-s_tp.fits to mastDownload/TESS/206409997/tess2018206045859-s0001-0000000206409997-0120-s_tp.fits ...
|=========================================================================================================================================================================================================================================|  49M/ 49M (100.00%)         2s
```

### CODE REVIEW

File: astroquery/mast/core.py [hunk 1]
```diff
@@ -1493,85 +1496,70 @@ def _shib_download_curl_script(self, products, out_dir):
...
+
+    def disable_s3_dataset(self):
         """
-        Disables downloading HST public files from S3 instead of MAST
+        Disables downloading public files from S3 instead of MAST
         """
         self._boto3 = None
         self._botocore = None
 
+    @deprecated(since="11/9/18", alternative="get_s3_uris")
```
Comment (bsipocz) [inline astroquery/mast/core.py]: please use an international date format, preferably ISO (YYYY-MM-DD), this it super confusing.
or just add v0.3.9 for the release, I think we need to tag a version sometimes soon.

Comment (cam72cam) [reply astroquery/mast/core.py]: Fixed :+1:

File: astroquery/mast/core.py [hunk 2]
```diff
@@ -1493,85 +1496,70 @@ def _shib_download_curl_script(self, products, out_dir):
                           "URL": [url]})
         return manifest
 
+    @deprecated(since="v0.3.9", alternative="enable_s3_dataset")
     def enable_s3_hst_dataset(self):
+        return self.enable_s3_dataset()
+
+    def enable_s3_dataset(self):
```
[the remark this answers is not in the document]
Comment (cam72cam) [reply astroquery/mast/core.py]: Done

[the remark this answers is not in the document]
Comment (arfon) [reply astroquery/mast/core.py]: 💖

**Comment by bsipocz** [comment]
Given the merge of #1268, this run into a conflict, but I've rebased and now merging (as the tests were passing previously).

**Comment by bsipocz** [comment]
Thank you @cam72cam!

**Comment by bsipocz** [comment]
Actually I've messed up the PR, and pushed up here before all the commits were included in the rebase. #1276 fixes that error. 

I'm sorry about this mess.

**Pull request merged**